diff -up ibus-chewing-1.4.3-Source/CMakeLists.txt~ ibus-chewing-1.4.3-Source/CMakeLists.txt
--- ibus-chewing-1.4.3-Source/CMakeLists.txt~	2012-11-26 09:25:20.000000000 +0200
+++ ibus-chewing-1.4.3-Source/CMakeLists.txt	2013-02-21 00:02:13.546391044 +0200
@@ -35,7 +35,7 @@ IBM, Gin-Yeah, Eten, Eten 26, Hsu, Dvora
 
 Chewing also support toned Hanyu pinyin input.")
 
-SET(CMAKE_C_FLAGS "-Wall")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 SET(AUTHORS "Peng Huang, Ding-Yi Chen")
 SET(MAINTAINER "Ding-Yi Chen <dchen at redhat.com>")
