diff -up ./src/IBusChewingEngine.gob.rhbz1062133 ./src/IBusChewingEngine.gob
--- ./src/IBusChewingEngine.gob.rhbz1062133	2014-12-17 14:50:06.991279350 +1000
+++ ./src/IBusChewingEngine.gob	2014-12-17 14:50:14.933243944 +1000
@@ -1101,7 +1101,9 @@ class IBus:Chewing:Engine from IBus:Engi
 
 	ibus_chewing_engine_set_status_flag(self,
 		ENGINE_STATUS_PROPERTIES_REGISTERED);
-
+	
+	/* Commit empty text to fix 1062133 */
+	ibus_engine_commit_text(engine, SELF_GET_CLASS(self)->emptyText);
 //	self_update(self);
 	IBUS_CHEWING_LOG(4,"[I4] focus_in(): return");
     }
